cPP_DSC.adoc

```diff
@@ -925,88 +929,80 @@ _Application Note {counter:remark_count}_:: _The HMAC minimum key sizes in the t
 |List of Standards
 
 |RSA-PKCS
-|RSASSA-PKCS1-v1_5 using {empty}[*selection*: [.underline]#SHA-256, SHA-384, SHA-512, SHA3-256, SHA3-384, SHA3-512#] 
-|{empty}[*selection*: [.underline]#2048, 3072, 4096#] bits 
-|{empty}[*selection*: [.underline]#RFC 8017, PKCS #1 v2.2 (Section 8.2); FIPS PUB 186-5 (Section 5.4)#] [RSASSA-PKCS1-v1_5]
+|RSASSA-PKCS1-v1_5
+|Modulus of size [selection: 2048, 3072, 4096] bits, hash or XOF [selection: SHA-256, SHA-384, SHA-512, SHA3-256, SHA3-384, SHA3-512, SHAKE-128, SHAKE-256]
+|[selection: RFC 8017, PKCS #1 v2.2 (Section 8.2), FIPS PUB 186-5 (Section 5.4)] [RSASSA-PKCS1-v1_5]
 
-{empty}[*selection*: [.underline]#ISO/IEC 10118-3:2018, FIPS PUB 180-4, FIPS PUB 202#] [SHA]
+[selection: ISO/IEC 10118-3:2018, FIPS PUB 180-4 [SHA2], FIPS PUB 202] [SHA3, SHAKE]
 
 |RSA-PSS
-|RSASSA-PSS using {empty}[*selection*: [.underline]#SHA-256, SHA-384, SHA-512, SHA3-256, SHA3-384, SHA3-512#]
-|{empty}[*selection*: [.underline]#2048, 3072, 4096#] bits
-|{empty}[*selection*: [.underline]#RFC 8017, PKCS#1v2.2 (Section 8.1); FIPS PUB 186-5 (Section 5.4)]# [RSASSA-PSS]
-
-{empty}[*selection*: [.underline]#ISO/IEC 10118-3:2018, FIPS PUB 180-4, FIPS PUB 202#] [SHA]
-
-|ECDSA
-|ECDSA on {empty}[*selection*: [.underline]#brainpoolP256r1, brainpoolP384r1, brainpoolP512r1, NIST P-256, NIST P-384, NIST P-521#] using {empty}[*selection*: [.underline]#SHA-256, SHA-384, SHA-512, SHA3-256, SHA3-384, SHA3-512#]
-|{empty}[*selection*: [.underline]#256, 384, 512, 521#] bits
-|{empty}[*selection*: [.underline]#ISO/IEC 14888-3:2018 (Sub Clause 6.6), FIPS PUB 186-5 (Sections 6.3.1, 6.4.1)]# [ECDSA]
+|RSASSA-PSS
+|Modulus of size [selection: 2048, 3072, 4096] bits, hash or XOF [selection: SHA-256, SHA-384, SHA-512, SHA3-256, SHA3-384, SHA3-512, SHAKE-128, SHAKE-256]
+|[selection: RFC 8017, PKCS#1v2.2 (Section 8.1); FIPS PUB 186-5 (Section 5.4)] [RSASSA-PSS]
 
-{empty}[*selection*: [.underline]#RFC 5639 (Section 3) [Brainpool Curves], NIST SP-800-186-5 (Section 3) [NIST Curves]#]
+[selection: ISO/IEC 10118-3:2018, FIPS PUB 180-4 [SHA2], FIPS PUB 202] [SHA3]
 
-{empty}[*selection*: [.underline]#ISO/IEC 10118-3:2018, FIPS PUB 202#] [SHA]
 
-|Det-ECDSA
-|Deterministic ECDSA on {empty}[*selection*: brainpoolP256r1, brainpoolP384r1, [.underline]#brainpoolP512r1, NIST P-256, NIST P-384, NIST P-521#] using {empty}[*selection*: [.underline]#SHA-256, SHA-384, SHA-512, SHA3-256, SHA3-384, SHA3-512#]
-|{empty}[*selection*: [.underline]#256, 384, 512, 521#] bits
-|{empty}[*selection*: [.underline]#ISO/IEC 14888-3:2018 (Sub Clause 6.6), FIPS PUB 186-5 (Sections 6.3.2, 6.4.1)]# [Deterministic ECDSA]
+|ECDSA
+|ECDSA
+|Elliptic curve [selection: P-256, brainpoolP256r1, P-384, brainpoolP384r1, P-521, brainpoolP512r1], per-message secret number generation [selection: extra random bits, rejection sampling, deterministic] and hash or XOF function using [selection: SHA-256, SHA-384, SHA-512, SHA3-256, SHA3-384, SHA3-512, SHAKE-128, SHAKE-256]
```

Comment:
@slpotte

Reply:
I will defer to the ISO document, to which I don't have access.  If allowed in ISO, then I would vote to keep it here and schemes can choose or limit as they see fit for their purposes.  Of course, there's this issue of evaluation activities, but I'm getting ahead of myself.

Reply:
@slpotte yeah, best to defer to the ISO. I had a quick look and it seems like ISO doesn't actually specify anything, just that it has to be a "cryptographically secure hash function". But maybe I missed something.

Reply:
Since NIST has determined that SHAKE128 and SHAKE256 are extended output functions and NOT hash functions, this furthers the case for removing them from here.  I will remove them from the catalog.

Reply:
I'm also removing them from the Korean rows as well.

Reply:
@slpotte
> Since NIST has determined that SHAKE128 and SHAKE256 are extended output functions and NOT hash functions, this furthers the case for removing them from here. I will remove them from the catalog.

SHAKE128 and SHAKE256 can still remain when "extra random bits" or "rejection sampling" are selected. It's only when "deterministic" is selected that there's an issue. If you remove SHAKE128 and SHAKE256 from the row entirely, you do remove the options for "extra random bits" and "rejection sampling". Rather than removing it from the rows, perhaps an application note is more appropriate?

> I'm also removing them from the Korean rows as well.

I didn't check the ISO for that one, I suggest waiting for the Koreans before dropping anything from the catalog.

Reply:
I would leave them in given that this is a list of all possible options that may be used based on the other selections.